web/components/admin/AssessmentPage/ScorePanel/Criteria.js

```diff
@@ -0,0 +1,54 @@
+const criteria = [
+  {
+    name: 'githubPersonalWebsite',
+    title: 'Github/Personal Website',
+  },
+  {
+    name: 'resumeLinkedin',
+    title: 'Resume/LinkedIn',
+  },
+  {
+    name: 'writtenResponse1',
+    title: "What is a project you're interested in building?",
+  },
+  {
+    name: 'writtenResponse2',
+    title: "Tell us about a recent project you’ve worked on that you're proud of!",
+  },
+];
+
+// Determines the weight ratio of the scores between
+//  1. the total score of the portfolio criteria
+//  2. the total score of the longAnswers criteria
+// +-----------------------+----------------------------------+
+// | Portfolio score (/10) | Portfolio / long answers scaling |
+// +-----------------------+----------------------------------+
+// | 1                     | 50/50                            |
+// | 2-5                   | 70/30                            |
+// | 6-8                   | 80/20                            |
+// | 9-10                  | 90/10                            |
+// +-----------------------+----------------------------------+
+const weightRatio = {
+  1: 0.5,
+  2: 0.7,
+  3: 0.7,
+  4: 0.7,
+  5: 0.7,
+  6: 0.8,
+  7: 0.8,
+  8: 0.8,
+  9: 0.9,
+  10: 0.9,
+};
+
+// calculates the final score given an map of criteria name to score value
+const calculateFinalScore = (scores) => {
+  const portfolioScore = scores.githubPersonalWebsite + scores.resumeLinkedin;
+  const longAnswersScore = scores.writtenResponse1 + scores.writtenResponse2;
+  const portfolioWeight = weightRatio[portfolioScore];
+  const longAnswersWeight = 1 - portfolioWeight;
+  const finalScore = portfolioWeight * portfolioScore + longAnswersWeight * longAnswersScore;
+  return Math.round(finalScore * 10) / 10; // round to one decimal place
```

Comment:
It's to ensure that it displays and saves the final score by one decimal place. I tried removing the *10 and /10 but it would display and save the final score as a whole number

Reply:
In general I think it's better to store the exact value in the database/back-end and just round the number when showing it in the UI/front-end. 

In addition to considering the sorting/tie break behaviour that @StanfordLin mentioned, we should also think about filtering - if an applicant's the precise final score was 6.49, and it got rounded to 6.5, should it qualify when we're filtering applicants whose final score is at least 6.5?

Reply:
If the decimal place is used in filtering and possibly tie breaking then we should show exactly how much of a decimal place matters to the admin. We have to decide whether we want to round to two or one decimal place, and we should keep it consistent between the DB and the frontend. I don't think admins should need to go into the database to break ties and a hacker listed as 6.5 in the ui should definitely not be dropped during sorting, even if their db score is below 6.5.